app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionaryListFragment.java

```diff
@@ -105,23 +143,43 @@ public static TreeSet<String> getUserDictionaryLocalesSet(final Activity activit
      */
     protected void createUserDictSettings(final PreferenceGroup userDictGroup) {
         userDictGroup.removeAll();
-        final TreeSet<String> localeSet = UserDictionaryList.getUserDictionaryLocalesSet(requireActivity());
-        if (localeSet == null) {
+        final TreeSet<String> enabledUserDictionary = getUserDictionaryLocalesSet(requireActivity());
+        // List of system language
+        final List<Locale> enabledSystemLocale = SubtypeSettingsKt.getSystemLocales();
+        // List of all enabled system languages
+        Set<String> systemLocales = new HashSet<>();
```

Comment:
Why don't you just put everything in a `LinkedHashSet`? As far as I understand you use the two sets only for the order, right?
Then using `LinkedHashSet` lets you avoid removing duplicates and adding prefs from two different sets.

Reply:
Don't mind this piece of code because I was about to delete it entirely.
Indeed, if there are no words in these system dictionaries, why display them? It's very confusing.

It won't bother the user at all, as these languages are always displayed in the drop-down menu when a word is added.

Only the "For all languages" dictionary will appear as the initial view when no word is added.

Do you agree with this way of thinking?

Reply:
> I would be ok with showing empty dictionaries, simply because the default (system) personal dictionary settings behave the same way.
>
> But removing / not showing empty dictionaries is definitely not wrong.

I still can't decide...
Anyway, I didn't know about LinkedHashSet and it's really great.
Done with the commit : 3dd6150